controllers/nutanixmachine_controller.go

```diff
@@ -503,6 +504,15 @@ func (r *NutanixMachineReconciler) getOrCreateVM(rctx *nctx.MachineContext) (*nu
 			Categories:  categories,
 		}
 
+		vmMetadataPtr := &vmMetadata
+		err = r.addVMToProject(rctx, vmMetadataPtr)
+		if err != nil {
+			errorMsg := fmt.Errorf("error occurred while trying to add VM %s to project: %v", vmName, err)
+			rctx.SetFailureStatus(capierrors.CreateMachineError, errorMsg)
+			klog.Errorf("%s %v", rctx.LogPrefix, errorMsg)
```

Comment:
is project mandatory to be provided. should it be optional?

Reply:
it is optional